meltano_map_transform/mapper.py

```diff
@@ -0,0 +1,148 @@
+            validate_config=validate_config,
+        )
+
+        self.mapper = PluginMapper(plugin_config=dict(self.config), logger=self.logger)
+
+    def map_schema_message(
+        self,
+        message_dict: dict,
+    ) -> Generator[singer.Message, None, None]:
+        """Map a schema message according to config.
+
+        Args:
+            message_dict: A SCHEMA message JSON dictionary.
+
+        Yields:
+            Transformed schema messages.
+        """
+        self._assert_line_requires(message_dict, requires={"stream", "schema"})
+
+        stream_id: str = message_dict["stream"]
+        self.mapper.register_raw_stream_schema(
+            stream_id,
+            message_dict["schema"],
+            message_dict.get("key_properties", []),
+        )
+        for stream_map in self.mapper.stream_maps[stream_id]:
+            schema_message = singer.SchemaMessage(
+                stream_map.stream_alias,
+                stream_map.transformed_schema,
+                stream_map.transformed_key_properties,
+                message_dict.get("bookmark_keys", []),
+            )
+            yield schema_message
+
+    def map_record_message(
+        self,
+        message_dict: dict,
+    ) -> Generator[singer.Message, None, None]:
+        """Map a record message according to config.
+
+        Args:
+            message_dict: A RECORD message JSON dictionary.
+
+        Yields:
+            Transformed record messages.
+        """
+        self._assert_line_requires(message_dict, requires={"stream", "record"})
+
+        stream_id: str = message_dict["stream"]
+        for stream_map in self.mapper.stream_maps[stream_id]:
+            mapped_record = stream_map.transform(message_dict["record"])
+            if mapped_record is not None:
+                record_message = singer.RecordMessage(
+                    stream=stream_map.stream_alias,
+                    record=mapped_record,
+                    version=message_dict.get("version"),
+                    time_extracted=utc_now(),
+                )
+                self.logger.info(stream_map.stream_alias)
+                yield record_message
+
+    def map_state_message(self, message_dict: dict) -> List[singer.Message]:
+        """Do nothing to the message.
+
+        Args:
+            message_dict: A STATE message JSON dictionary.
+
+        Returns:
+            The same state message
+        """
+        return [singer.StateMessage(value=message_dict["value"])]
+
+    def map_activate_version_message(self, message_dict: dict) -> List[singer.Message]:
+        """Do nothing to the message.
+
+        Args:
+            message_dict: An ACTIVATE_VERSION message JSON dictionary.
+
+        Returns:
+            The same state message
+        """
+        return [
+            singer.ActivateVersionMessage(
+                stream=message_dict["stream"],
+                version=message_dict["version"],
+            )
+        ]
```

Comment:
Because the stream map can generate multiple aliases/projections of the source stream, I think we should loop through the stream maps and use `stream_map.stream_alias` in place of `message_dict["stream"]` here.

I think you can use the same loop as above:

```py
for stream_map in self.mapper.stream_maps[stream_id]:
    ...
```

Reply:
Makes sense! TIL about `__source__` and `__alias__` 😃